### PR TITLE
Bump black and formatting to new version

### LINKED ISSUE #4931: Update black and other checks to latest
Black was updated with major changes, which requires to rerun it over the whole project.

At the same time, we could take the time to update flake8, isort, etc. from pre-commit.

TBD:
- [ ] Update black to latest
- [ ] Update black version in `tox.ini`
- [ ] Update `.pre-commit-config.yaml` by running `pre-commit autoupdate`

### LINKED ISSUE #5013: Black should use target version py38
Since we only use python version specified in dockerfile, we can format code using only one target.

Warning: this will trigger style changes all over the code.

### PR DESCRIPTION
**Pull request by krzysztofwolski**
I want to merge this change because...
...Black no longer crashes on Windows machines with more than 61 cores and allows for hanging `,`

Closes #4931 (done)
Closes #5013 (done)

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for f8ed7294b141d248d28ca545b1b4e43f6025ea0c (krzysztofwolski/saleor @ bump-black)
Base comparison is da715b1be241a8f456221a7771ec56491e3d3f9f.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         10	         10	              0
  complete checkout                          	          8	          8	              0
  create checkout                            	         50	         50	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         15	         15	              3
  retrieve product attributes                	          9	          9	              0

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         15	         15	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

**Comment by maarcingebala** [comment]
@krzysztofwolski Does this PR solve this issue by any chance? https://github.com/mirumee/saleor/issues/5013

**Comment by maarcingebala** [comment]
Or this one: https://github.com/mirumee/saleor/issues/4931

**Comment by IKarbowiak** [comment]
What about `requirements.txt` and `requirements_dev.txt`? AFAIK it should be updated too.

**Comment by NyanKiyoshi** [comment]
> Or this one: #4931

Ain't this commit fixed it https://github.com/mirumee/saleor/commit/8fac1908ee27b97742644b05c8ad6ce9a28720d1#diff-b91f3d5bd63fcd17221b267e851608e8?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 repos:
   - repo: https://github.com/ambv/black
-    rev: 19.3b0
+    rev: 19.10b0
     hooks:
       - id: black
         language_version: python3
```
Comment (NyanKiyoshi) [inline .pre-commit-config.yaml]:
```suggestion
         language_version: python3.8
```

**Comment by krzysztofwolski** [comment]
@NyanKiyoshi interesting: 
```
An error has occurred: InvalidConfigError: 
==> File .pre-commit-config.yaml
=====> mapping values are not allowed here
  in "<unicode string>", line 6, column 26:
             language_version: python3.8
                             ^
Check the log at /Users/krzyh/.cache/pre-commit/pre-commit.log
```

**Pull request merged**